File: 4th Semester/Design and programming environments/Proiect C-Sharp Client-Server Sockets/AppPersistance/database/ParticipantDBRepository.cs
using AppModel;
using log4net;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AppPersistance.database
{
    public class ParticipantDBRepository : ParticipantRepository
    {

        private static readonly ILog log = LogManager.GetLogger("ParticipantDBRepository");
        public ParticipantDBRepository()
        {
            log.Info("Creating SortingTaskDbRepository");
        }
        public void Delete(int id)
        {
            IDbConnection con = DBUtils.getConnection();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "delete from Participant where ID=@id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);
                var dataR = comm.ExecuteNonQuery();
                if (dataR == 0) ;
                //throw new RepositoryException("No task deleted!");
            }
        }

        public IEnumerable<Participant> filterByName(string name)
        {
            log.InfoFormat("Entering findOne with value {0}", name);
            IDbConnection con = DBUtils.getConnection();
            IList<Participant> list = new List<Participant>();

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Participant where nume=@nume";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@nume";
                paramId.Value = name;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        int id1 = dataR.GetInt32(0);
                        string nume = dataR.GetString(1);
                        int varsta = dataR.GetInt32(2);
                        Participant participant = new Participant(varsta, nume);
                        participant.id = id1;
                        list.Add(participant);
                    }
                }
            }
            log.InfoFormat("Exiting findOne with value {0}", null);
            return list;
        }

        public IEnumerable<Participant> filterByVarsta(int varsta)
        {
            log.InfoFormat("Entering findOne with value {0}", varsta);
            IDbConnection con = DBUtils.getConnection();
            IList<Participant> list = new List<Participant>();

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Participant where nume=@varsta";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@varsta";
                paramId.Value = varsta;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        int id1 = dataR.GetInt32(0);
                        string nume = dataR.GetString(1);
                        int varsta2 = dataR.GetInt32(2);
                        Participant participant = new Participant(varsta2, nume);
                        participant.id = id1;
                        list.Add(participant);
                    }
                }
            }
            log.InfoFormat("Exiting findOne with value {0}", null);
            return list;
        }

        public IEnumerable<Participant> FindAll()
        {
            IDbConnection con = DBUtils.getConnection();
            IList<Participant> list = new List<Participant>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Participant";

                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {

                        int id = dataR.GetInt32(0);
                        string nume = dataR.GetString(1);
                        int varsta = dataR.GetInt32(2);
                        Participant participant = new Participant(varsta, nume);
                        participant.id = id;
                        list.Add(participant);
                    }
                }
            }

            return list;
        }

        public Participant FindOne(int id)
        {
            log.InfoFormat("Entering findOne with value {0}", id);
            IDbConnection con = DBUtils.getConnection();

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Participant where ID=@id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        int id1 = dataR.GetInt32(0);
                        string nume = dataR.GetString(1);
                        int varsta = dataR.GetInt32(2);
                        Participant participant = new Participant(varsta, nume);
                        participant.id = id1;
                        return participant;
                    }
                }
            }
            log.InfoFormat("Exiting findOne with value {0}", null);
            return null;
        }

        public Participant Save(Participant entity)
        {
            var con = DBUtils.getConnection();

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "insert into Participant (nume,varsta) values (@nume, @varsta)";

                var paramIDParticipant = comm.CreateParameter();
                paramIDParticipant.ParameterName = "@nume";
                paramIDParticipant.Value = entity.nume;
                comm.Parameters.Add(paramIDParticipant);

                var paramIDProba = comm.CreateParameter();
                paramIDProba.ParameterName = "@varsta";
                paramIDProba.Value = entity.varsta;
                comm.Parameters.Add(paramIDProba);

                var result = comm.ExecuteNonQuery();
                if (result == 0) ;
                //throw new RepositoryException("No task added !");
            }
            return entity;
        }

    }
}

File: 4th Semester/Design and programming environments/Project Java Client-Server/Persistance/src/main/java/app/persistance/repository/databases/ParticipantDBRepository.java
package app.persistance.repository.databases;

import app.persistance.repository.ParticipantRepository;
import app.persistance.repository.utils.JdbcUtils;
import domain.Participant;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class ParticipantDBRepository implements ParticipantRepository {
    private JdbcUtils dbUtils;

    private static final Logger logger = LogManager.getLogger();

    public ParticipantDBRepository(Properties props) {
        logger.info("Initializing CarsDBRepository with properties: {} ", props);
        dbUtils = new JdbcUtils(props);
    }

    @Override
    public Participant findOne(Integer id) {
        logger.traceEntry("selecting entity {}", id);
        Connection con = dbUtils.getConnection();
        Participant participant1 = new Participant();

        try (PreparedStatement preStmt = con.prepareStatement("select * from Participant where ID=?")) {
            preStmt.setInt(1, (Integer) id);
            ResultSet result = preStmt.executeQuery();
            int id1 = result.getInt("ID");
            String nume = result.getString("nume");
            int varsta = result.getInt("varsta");
            Participant participant = new Participant(nume, varsta);
            participant.setId(id1);
            participant1 = participant;
            logger.trace("Selected {} instance", result);
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB " + ex);
        }
        logger.traceExit();
        return participant1;
    }

    @Override
    public Iterable<Participant> findAll() {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<Participant> participants = new ArrayList<>();
        try (PreparedStatement preStmt = con.prepareStatement("select * from Participant")) {
            try (ResultSet result = preStmt.executeQuery()) {
                while (result.next()) {
                    int id = result.getInt("ID");
                    String nume = result.getString("nume");
                    int varsta = result.getInt("varsta");
                    Participant participant = new Participant(nume, varsta);
                    participant.setId(id);
                    participants.add(participant);
                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(participants);
        return participants;
    }


    @Override
    public Participant save(Participant entity) {
        Participant participant = (Participant) entity;
        logger.traceEntry("saving entity {}", entity);
        Connection con = dbUtils.getConnection();
        try (PreparedStatement preStmt = con.prepareStatement("insert into Participant (nume, varsta) values (?,?)")) {
            preStmt.setString(1, participant.getNume());
            preStmt.setInt(2, participant.getVarsta());
            int result = preStmt.executeUpdate();
            logger.trace("Saved {} instances", result);
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB " + ex);
        }
        logger.traceExit();
        return participant;
    }

    @Override
    public void delete(Integer id) {
        logger.traceEntry("deleting entity {}", id);
        Connection con = dbUtils.getConnection();
        try (PreparedStatement preStmt = con.prepareStatement("delete from Participant where ID=?")) {
            preStmt.setInt(1, (Integer) id);
            int result = preStmt.executeUpdate();
            logger.trace("Deleted {} instances", result);
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB " + ex);
        }
        logger.traceExit();
    }

    @Override
    public Participant update(Integer integer, Participant entity) {
        return null;
    }

    @Override
    public Iterable<Participant> filterByName(String name) {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<Participant> participants = new ArrayList<>();
        try (PreparedStatement preStmt = con.prepareStatement("select * from Participant where nume=?")) {
            preStmt.setString(1, name);
            try (ResultSet result = preStmt.executeQuery()) {
                while (result.next()) {
                    int id = result.getInt("ID");
                    int varsta = result.getInt("varsta");
                    Participant participant = new Participant(name, varsta);
                    participant.setId(id);
                    participants.add(participant);
                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(participants);
        return participants;
    }

    @Override
    public Iterable<Participant> filterByVarsta(int varsta) {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<Participant> participants = new ArrayList<>();
        try (PreparedStatement preStmt = con.prepareStatement("select * from Participant where varsta=?")) {
            preStmt.setInt(1, varsta);
            try (ResultSet result = preStmt.executeQuery()) {
                while (result.next()) {
                    int id = result.getInt("ID");
                    String name = result.getString("nume");
                    Participant participant = new Participant(name, varsta);
                    participant.setId(id);
                    participants.add(participant);
                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(participants);
        return participants;
    }
}
